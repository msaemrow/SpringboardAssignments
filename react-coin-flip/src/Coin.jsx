import React from "react"
import headsImg from './assets/coin-heads.jpg';
import tailsImg from './assets/coin-tails.jpg';

function Coin({coinSide}){
    if (!coinSide) {
        return (
            <h5>
                Click the button to start the game!
            </h5>
        );
    }

    const coinImg = coinSide === "heads" ? headsImg : tailsImg;

    return(
        <div className="Coin">
        <img className="Coin-image" src={coinImg} alt={coinSide} />
        </div>
    )
}

export default Coin;