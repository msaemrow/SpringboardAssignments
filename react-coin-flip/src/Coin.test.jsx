import { render } from "@testing-library/react";
import Coin from "./Coin";
import { expect, it } from "vitest";


it("loads Coin without crashing", () => {
    render(<Coin coinSide={null} />)
})

it("matches snapshot", () => {
    const { asFragment } = render(
        <Coin coidSide={null} />
    );
    expect(asFragment()).toMatchSnapshot();
})

