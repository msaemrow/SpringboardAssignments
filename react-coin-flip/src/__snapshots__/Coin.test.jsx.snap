// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h5>
    Click the button to start the game!
  </h5>
</DocumentFragment>
`;
